thoth/common/workflows.py

```diff
@@ -188,6 +188,162 @@ def get_workflow_template(
+        label_selector: str,
+    ) -> dict:
+        """Get Workflows in namespace."""
+        response = self.api.list_namespaced_workflows(namespace=namespace, label_selector=label_selector)
+        return response.to_dict()
+
+    def get_workflow_info(
+        self, namespace: str, name: str,
+    ):
+        """Get Workflow in namespace by name."""
+        workflow = self.get_workflow(namespace=namespace, name=name)
+        workflow_main = self._collect_workflow_info(workflow=workflow)
+        return workflow_main
+
+    def get_workflows_info(
+        self, label_selector: str, namespace: str
+    ) -> Dict[str, int]:
+        """Get workflows info from a specific namespace."""
+        workflows = self.get_workflows(namespace=namespace, label_selector=label_selector)
+        workflow_data = {}
+        for workflow in workflows["items"]:
+
+            workflow_main = self._collect_workflow_info(workflow=workflow)
+            
+            workflow_data[workflow['metadata']['name']] = workflow_main
+
+        return workflow_data
+
+    def _collect_workflow_info(self, workflow: dict) -> dict:
+        """Collect Workflow info."""
+        workflow_info = {}
+        workflow_info["name"] = workflow['metadata']['name']
+
+        started_at = workflow['status']['started_at']
+        workflow_info["started_at"] = started_at
+
+        finished_at = workflow['status']['finished_at']
+        workflow_info["finished_at"] = finished_at
+
+        duration = None
+        if started_at and finished_at:
+            duration = (finished_at - started_at).total_seconds()
+        workflow_info["duration"] = duration
+
+        workflow_info["phase"] = workflow['status']['phase']
+
+        tasks_names = self._collect_tasks_names(templates=workflow['spec']['templates'])
+        nodes = self._collect_workflows_tasks_info(nodes=workflow['status']['nodes'], tasks_names=tasks_names)
+        workflow_info["nodes"] = nodes
+
+        return workflow_info
+
+    @staticmethod
+    def _collect_tasks_names(templates: List[dict]):
+        """Collect tasks names from Workflow template."""
+        tasks_names = []
+
+        for template in templates:
+            # TODO: Extend to other structures beside dag and consider also steps
+            if 'dag' in template.keys():
+                for task in template['dag']['tasks']:
+                    tasks_names.append(task['name'])
+        return tasks_names
+
+    @staticmethod
+    def _collect_workflows_tasks_info(nodes: dict, tasks_names: List[str]) -> dict:
+        """Collect info about tasks in the Workflow."""
+        nodes_info = {}
+        if not nodes:
+            return nodes_info
+
+        for pod_id, node in nodes.items():
+            
+            if node['display_name'] in tasks_names:
+                nodes_info[pod_id] = {}
+                nodes_info[pod_id]['pod_id'] = pod_id
+                nodes_info[pod_id]['display_name'] = node['display_name']
```

Comment:
This entry can be initialized at the end with all the values to safe some CPU cycles:

```
                nodes_info[pod_id] = {
                  "pod_id": pod_id,
                  "display_name": node["display_name"],
                  "finished_at": finished_at,
                   ...
                }
```